### PR TITLE
Adding support for Forms Conversion to aem modernize tool

### PR DESCRIPTION
**Pull request by ujjwal5**
## Description

- Added an executor for Form conversion job.
- Form conversion job would support following form handling:
        **None** [applies an in place component conversion]
        **Restore** [Restore form to a state to last structure conversion and then apply component rules]
        **Copy to Target** [Copying forms from Source directory to Target directory and then applying component conversion.]

For "Copy to Target" option, we are also copying dam asset node to target folder specific to forms.

**NOTE**: I had to disable baseline check since a version upgrade was required in ConversionJob model. Will enable it once this is through bumping up the versions as well.

## Related Issue
https://github.com/adobe/aem-modernize-tools/issues/209

## Motivation and Context

Added in the mentioned issue above.

## How Has This Been Tested?

Added UTs for the form conversion executor.

## Screenshots (if appropriate):

https://github.com/adobe/aem-modernize-tools/assets/7971922/0cc895c8-29a5-47ff-81a9-89aacf3184d0

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -50,7 +50,7 @@ Bundle-SymbolicName: com.adobe.aem.aem-modernize-tools.core
                     </execution>
                 </executions>
             </plugin>
-            <plugin>
+            <!-- <plugin>
```
Comment (bstopp) [inline core/pom.xml:53]: This plugin is required to be used in order to validate that the versions are being set correctly based on changes.

File: core/src/main/java/com/adobe/aem/modernize/model/ConversionJob.java [hunk 2]
```diff
@@ -26,6 +24,7 @@
 import javax.inject.Inject;
 import javax.inject.Named;
 
+import com.adobe.aem.modernize.job.FormConversionJobExecutor;
```
Comment (bstopp) [inline core/src/main/java/com/adobe/aem/modernize/model/ConversionJob.java:27]: Order imports - this should be down with the other jobs.

File: core/src/test/java/com/adobe/aem/modernize/job/FormConversionJobExecutorTest.java [hunk 3]
```diff
@@ -0,0 +1,296 @@
+package com.adobe.aem.modernize.job;
```
Comment (bstopp) [inline core/src/test/java/com/adobe/aem/modernize/job/FormConversionJobExecutorTest.java:1]: This should be in a sub-package that context specific, e.g. `forms`

File: core/src/main/java/com/adobe/aem/modernize/job/FormConversionJobExecutor.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
+package com.adobe.aem.modernize.job;
```
Comment (bstopp) [inline core/src/main/java/com/adobe/aem/modernize/job/FormConversionJobExecutor.java:1]: This should be in a package that is context specific, e.g. `forms`

File: core/src/main/java/com/adobe/aem/modernize/servlet/AbstractRulesServlet.java [hunk 5]
```diff
@@ -103,7 +101,6 @@ protected Resource getPageContent(@NotNull Map<String, String[]> params, @NotNul
     }
     return pageContent;
   }
-
```
Comment (bstopp) [inline core/src/main/java/com/adobe/aem/modernize/servlet/AbstractRulesServlet.java:106]: See above

File: core/src/main/java/com/adobe/aem/modernize/servlet/AbstractRulesServlet.java [hunk 6]
```diff
@@ -68,7 +67,6 @@ protected void doGet(@NotNull SlingHttpServletRequest request, @NotNull SlingHtt
       writeResponse(response, SC_OK, data);
       return;
     }
-
```
Comment (bstopp) [inline core/src/main/java/com/adobe/aem/modernize/servlet/AbstractRulesServlet.java:71]: Fix formatting, don't remove spacing between methods.

File: core/src/main/java/com/adobe/aem/modernize/servlet/AbstractRulesServlet.java [hunk 7]
```diff
@@ -172,10 +168,8 @@ private void writeResponse(SlingHttpServletResponse response, int code, Response
 
   @NotNull
   protected abstract Set<String> listPaths(@NotNull Map<String, String[]> requestParameters, @NotNull Page page);
-
```
Comment (bstopp) [inline core/src/main/java/com/adobe/aem/modernize/servlet/AbstractRulesServlet.java:175]: See above

**Review by bstopp — CHANGES_REQUESTED** [review]
Biggest thing is that the forms code and test resources need to be in sub folders/packages

File: core/pom.xml (hunk 1 above)
Comment (ujjwal5) [reply core/pom.xml]: Uncommented this code and bumped version for all packages.

File: core/src/main/java/com/adobe/aem/modernize/model/ConversionJob.java (hunk 2 above)
Comment (ujjwal5) [reply core/src/main/java/com/adobe/aem/modernize/model/ConversionJob.java]: done

File: core/src/test/java/com/adobe/aem/modernize/job/FormConversionJobExecutorTest.java (hunk 3 above)
Comment (ujjwal5) [reply core/src/test/java/com/adobe/aem/modernize/job/FormConversionJobExecutorTest.java]: done

File: core/src/main/java/com/adobe/aem/modernize/job/FormConversionJobExecutor.java (hunk 4 above)
Comment (ujjwal5) [reply core/src/main/java/com/adobe/aem/modernize/job/FormConversionJobExecutor.java]: done

File: core/src/main/java/com/adobe/aem/modernize/servlet/AbstractRulesServlet.java (hunk 5 above)
Comment (ujjwal5) [reply core/src/main/java/com/adobe/aem/modernize/servlet/AbstractRulesServlet.java:106]: added back spacing between methods.

File: core/src/main/java/com/adobe/aem/modernize/servlet/AbstractRulesServlet.java (hunk 6 above)
Comment (ujjwal5) [reply core/src/main/java/com/adobe/aem/modernize/servlet/AbstractRulesServlet.java:71]: done

**Comment by ujjwal5** [comment]
> Biggest thing is that the forms code and test resources need to be in sub folders/packages

Moved forms code to form specific folders/packages.

File: core/src/main/java/com/adobe/aem/modernize/component/package-info.java [hunk 8]
```diff
@@ -1,4 +1,4 @@
-@org.osgi.annotation.versioning.Version("2.1.0")
+@org.osgi.annotation.versioning.Version("2.2.0")
```
[the remark this answers is not in the document]
Comment (ujjwal5) [reply core/src/main/java/com/adobe/aem/modernize/component/package-info.java]: fixed

File: core/src/main/java/com/adobe/aem/modernize/policy/package-info.java [hunk 9]
```diff
@@ -1,4 +1,4 @@
-@org.osgi.annotation.versioning.Version("2.1.0")
+@org.osgi.annotation.versioning.Version("2.2.0")
```
[the remark this answers is not in the document]
Comment (ujjwal5) [reply core/src/main/java/com/adobe/aem/modernize/policy/package-info.java]: fixed

File: core/src/main/java/com/adobe/aem/modernize/rule/package-info.java [hunk 10]
```diff
@@ -1,4 +1,4 @@
-@org.osgi.annotation.versioning.Version("2.1.0")
+@org.osgi.annotation.versioning.Version("2.2.0")
```
[the remark this answers is not in the document]
Comment (ujjwal5) [reply core/src/main/java/com/adobe/aem/modernize/rule/package-info.java]: fixed

File: core/src/main/java/com/adobe/aem/modernize/structure/package-info.java [hunk 11]
```diff
@@ -1,4 +1,4 @@
-@org.osgi.annotation.versioning.Version("2.1.0")
+@org.osgi.annotation.versioning.Version("2.2.0")
```
[the remark this answers is not in the document]
Comment (ujjwal5) [reply core/src/main/java/com/adobe/aem/modernize/structure/package-info.java]: fixed

**Review by bstopp — APPROVED** [review]

**Pull request merged**